src/CostMatrix.cpp

```diff
@@ -0,0 +1,63 @@
+//
+// Created by mikita on 17.07.19.
+//
+
+#include "CostMatrix.h"
+
+bool CostMatrix::readFromFile(const std::string &filename)
+{
+    std::ifstream inFile;
+
+    inFile.open(filename);
+    if (!inFile)
+    {
+        std::cout << "Unable to open file " << filename << std::endl;
+        return false;
+    }
+
+    inFile >> this->size;
+    std::vector<std::vector<int>> matrix(size, std::vector<int>(size, 0));
+
+    try
+    {
+        for (int i = 0; i < this->size; ++i)
+        {
+            for (int j = 0; j < this->size; ++j)
+            {
+                inFile >> matrix[i][j];
+            }
+        }
+    }
+    catch (std::exception e)
```

Comment:
catching general exception is usually a sign of not knowing what could really be wrong. Replace with subclass or remove completely and replace with other error check.